src/library/autodj/autodjprocessor.cpp

```diff
@@ -859,17 +864,31 @@ void AutoDJProcessor::playerPositionChanged(DeckAttributes* pAttributes,
             double transitionProgress = (thisPlayPosition - thisDeck->fadeBeginPos) /
                     (thisDeck->fadeEndPos - thisDeck->fadeBeginPos);
             double transitionStep = transitionProgress - m_transitionProgress;
+
+            if ((transitionProgress == 0.0) &&
+                    (m_crossfaderStartCenter)) {
+                setCrossfader(0.0); // So we start at the mid point before we make any fader moves.
+            }
             if (transitionStep > 0.0) {
                 // We have made progress.
                 // Backward seeks pause the transitions; forward seeks speed up
                 // the transitions. If there has been a seek beyond endPos, end
                 // the transition immediately.
-                double remainingCrossfader = crossfaderTarget - currentCrossfader;
-                double adjustment = remainingCrossfader /
-                        (1.0 - m_transitionProgress) * transitionStep;
-                // we move the crossfader linearly with
-                // movements in this track's play position.
-                setCrossfader(currentCrossfader + adjustment);
+                if ((m_crossfaderStartCenter) &&
+                        (transitionProgress <=
+                                ((thisDeck->fadeEndPos -
+                                         thisDeck->fadeBeginPos) /
+                                        2))) {
+                    // We want the cross fader to remain in the middle.
```

Comment:
This will constantly fight with the user. I think it is better to just not touch the crossfadere here. It will feel naturally if we fall back to the default mode when xfader is no longer in the expected center position.

Reply:
This should now be:

`                 transitionProgress <= std::midpoint(thisDeck->fadeBeginPos, thisDeck->fadeEndPos) {`

to make it clearer what is going on.  

In my testing, neither constantly fights the user.  If the user moves the fader it goes there and completes the transition.

Reply:
I can confirm that this readjust the user tweaks in some conditions. It happens because you compare here fractions of the full track length with fractions of the transition. I think the best solution is to move this code to Line 823 as 
```
                if (m_crossfaderStartCenter) { 
                    setCrossfader(0.0);
                } else if (thisDeck->fadeBeginPos >= thisDeck->fadeEndPos) {
                    setCrossfader(thisDeck->isLeft() ? 1.0 : -1.0);
                }
```

Reply:
Please show the conditions.  I can't find any in my testing.

Reply:
I've had a play around with this.  And I think you are correct.  It seems you only need to set the position of the crossfader to the middle once.  And the best position would be just before the other desk starts playing.  So I would propose this at around line 820:

```
                if (m_crossfaderStartCenter) {
                    setCrossfader(0.0);
                } else if (thisDeck->fadeBeginPos >= thisDeck->fadeEndPos) {
                        setCrossfader(thisDeck->isLeft() ? 1.0 : -1.0);
                }

                if (!otherDeckPlaying) {
                    otherDeck->play();
                }
``` 
The reason is that I noticed that occasionally the audio seems to just miss the very start of a track (I noticed it on a cold vocal start).  This may have been a glitch in my system, so I cannot attest to this, but it seems the best thing to do in any event.